services/galley/test/integration/API/Teams.hs

```diff
@@ -559,31 +559,89 @@ testAddTeamConvWithUsers = do
 
 testAddTeamMemberToConv :: TestM ()
 testAddTeamMemberToConv = do
-    owner <- Util.randomUser
+    personalUser <- Util.randomUser
+    ownerT1 <- Util.randomUser
     let p = Util.symmPermissions [DoNotUseDeprecatedAddRemoveConvMember]
-    mem1 <- newTeamMember' p <$> Util.randomUser
-    mem2 <- newTeamMember' p <$> Util.randomUser
-    mem3 <- newTeamMember' (Util.symmPermissions []) <$> Util.randomUser
+    mem1T1 <- newTeamMember' p <$> Util.randomUser
+    mem2T1 <- newTeamMember' p <$> Util.randomUser
+    mem3T1 <- newTeamMember' (Util.symmPermissions []) <$> Util.randomUser
+    mem4T1 <- newTeamMember' (Util.symmPermissions []) <$> Util.randomUser
 
-    Util.connectUsers owner (list1 (mem1^.userId) [mem2^.userId, mem3^.userId])
-    tid <- Util.createTeam "foo" owner [mem1, mem2, mem3]
+    ownerT2 <- Util.randomUser
+    mem1T2 <- newTeamMember' p <$> Util.randomUser
 
-    -- Team owner creates new regular team conversation:
-    cid <- Util.createTeamConv owner tid [] (Just "blaa") Nothing Nothing
+    Util.connectUsers ownerT1 (list1 (mem1T1^.userId) [mem2T1^.userId, mem3T1^.userId, ownerT2, personalUser])
+    tidT1 <- Util.createTeam "foo" ownerT1  [mem1T1, mem2T1, mem3T1]
+    tidT2 <- Util.createTeamInternal "foo" ownerT2
+    _     <- Util.addTeamMemberInternal tidT2 mem1T2
+
+    -- Team owners create new regular team conversation:
+    cidT1 <- Util.createTeamConv ownerT1 tidT1 [] (Just "blaa") Nothing Nothing
+    cidT2 <- Util.createTeamConv ownerT2 tidT2 [] (Just "blaa") Nothing Nothing
+    cidPersonal <- decodeConvId <$> Util.postConv personalUser  [] (Just "blaa") [] Nothing Nothing
 
     -- NOTE: This functionality was _changed_ as there was no need for it...
-    -- Team member 1 (who is *not* a member of the new conversation)
-    -- can *not* add other team members without requiring a user connection
+    -- mem1T1 (who is *not* a member of the new conversation) can *not* add other team members
     -- despite being a team member and having the permission `DoNotUseDeprecatedAddRemoveConvMember`.
-    Util.assertNotConvMember (mem1^.userId) cid
-    Util.postMembers (mem1^.userId) (list1 (mem2^.userId) []) cid !!! const 404 === statusCode
-    Util.assertNotConvMember (mem2^.userId) cid
+    Util.assertNotConvMember (mem1T1^.userId) cidT1
+    Util.postMembers (mem1T1^.userId) (list1 (mem2T1^.userId) []) cidT1 !!! const 404 === statusCode
+    Util.assertNotConvMember (mem2T1^.userId) cidT1
 
-    -- OTOH, team member 3 _can_ add another team member despite lacking the required team permission
+    -- OTOH, mem3T1 _can_ add another team member despite lacking the required team permission
     -- However, all users are admins by default
-    Util.assertConvMember owner cid
-    Util.postMembers owner (list1 (mem2^.userId) []) cid !!! const 200 === statusCode
-    Util.assertConvMember (mem2^.userId) cid
+    Util.assertConvMember ownerT1 cidT1
```

Comment:
```suggestion
    Util.assertNotConvMember (mem2T1^.userId) cidT1
```

(or maybe both?)

Reply:
Ah no; if u look better, we're just ensuring `ownerT1` is a member of the conv before adding other members